sotodlib/tod_ops/sub_polyf.py

```diff
@@ -31,58 +42,147 @@ def subscan_polyfilter(aman, degree, signal=None, exclude_turnarounds=False, mas
-            _ = flags.get_turnaround_flags(aman)
+            _ = flags.get_turnaround_flags(aman,truncate=True)
         valid_scan = np.logical_and(np.logical_or(aman.flags["left_scan"].mask(), aman.flags["right_scan"].mask()),
                                     ~aman.flags["turnarounds"].mask())
         subscan_indices = _get_subscan_range_index(valid_scan)
     else:
         if ("left_scan" not in aman.flags):
             logger.warning('aman does not have left/right scan. `sotodlib.flags.get_turnaround_flags` will be ran with default parameters')
-            _ = flags.get_turnaround_flags(aman)
+            _ = flags.get_turnaround_flags(aman,truncate=True)
         subscan_indices_l = _get_subscan_range_index(aman.flags["left_scan"].mask())
         subscan_indices_r = _get_subscan_range_index(aman.flags["right_scan"].mask())
         subscan_indices = np.vstack([subscan_indices_l, subscan_indices_r])
         subscan_indices= subscan_indices[np.argsort(subscan_indices[:, 0])]
-    
+
+    signal = aman[signal_name]
+
+    if not(in_place):
+        signal = signal.copy()
+
     if mask is None:
         mask_array = np.zeros(aman.samps.count, dtype=bool)
     elif type(mask) is str:
         mask_array = aman.flags[mask].mask()
     else:
         mask_array = mask.mask()
+    if exclusive is False:
+        mask_array = ~mask_array
+        
     is_matrix = len(mask_array.shape) > 1
-    
-    t = aman.timestamps - aman.timestamps[0]
-    for i_det in range(aman.dets.count):
-        if is_matrix:
-            each_det_mask = mask_array[i_det]
-        else:
-            each_det_mask = mask_array
-
-        for start, end in subscan_indices:
-            if np.count_nonzero(~each_det_mask[start:end+1]) < degree:
-                # If degree of freedom is lower than zero, just subtract mean
-                signal[i_det, start:end+1] -= np.mean(signal[i_det, start:end+1])
+
+    if method not in ["polyfit","legendre"] :
+        raise ValueError("Only polyfit and legendre are acceptable.")
+
+    elif method == "polyfit":
+        t = aman.timestamps - aman.timestamps[0]
+        for i_det in range(aman.dets.count):
+            if is_matrix:
+                each_det_mask = mask_array[i_det]
             else:
-                t_mean = np.mean(t[start:end+1])
-                pars = np.ma.polyfit(
-                        np.ma.array(t[start:end+1]-t_mean, mask=each_det_mask[start:end+1]),
-                        np.ma.array(signal[i_det,start:end+1], mask=each_det_mask[start:end+1]),
-                        deg=degree)
+                each_det_mask = mask_array
+
+            for start, end in subscan_indices:
+                if np.count_nonzero(~each_det_mask[start:end+1]) < degree:
+                    # If degree of freedom is lower than zero, just subtract mean
+                    signal[i_det, start:end+1] -= np.mean(signal[i_det, start:end+1])
```

Comment:
I added this part while I was debugging the code.
Actually, I found that I had needed this part because of bug.
I think I can remove this part.
Are you agree with this @tterasaki ?

Reply:
How about leave this and add warning?